Controllers/StudentsController.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Mvc;
+using StudentService.Data;
+using StudentService.Models;
+
+namespace StudentService.Controller
+{
+    [Route("api/{controller}")]
+    [ApiController]
+    public class StudentsController : ControllerBase
+    {
+        private readonly IStudentRepo _repository;
+
+        public StudentsController(IStudentRepo repository)
+        {
+            _repository = repository;
+        }
+
+        //POST api/students
+        [HttpPost]
+        public ActionResult <Student> CreateStudent(Student student){
+            if(student == null){
+                return BadRequest();
+            }
+            _repository.CreateStudent(student);
```

Comment:
You shouldn't have repository directly in the controller

Reply:
How better could I have done it?

Reply:
I have added service layer for the application logic.